floodpipeline/forecast.py

```diff
@@ -0,0 +1,601 @@
+                        discharge_data_unit.discharge_ensemble,
+                    )
+
+                    #likelihood = 1 # this is a for deterministic forecast, so likelihood is 1
+                    likelihood= sum(threshold_checks) / len(discharge_data_unit.discharge_ensemble)
+
+                    return_period = threshold["return_period"]
+
+                    likelihood_per_return_period[return_period] = likelihood
+
+                    forecasts.append(
+                        FloodForecast(
+                            return_period=return_period, likelihood=likelihood
+                        )
+                    )
+
+                # determine if triggered and the corresponding return period
+                triggered = (
+                    True
+                    if likelihood_per_return_period[trigger_on_return_period]
+                    >= trigger_on_minimum_probability
+                    and lead_time <= trigger_on_lead_time
+                    else False
+                )
+                return_period = next(
+                    (
+                        key
+                        for key, value in reversed(likelihood_per_return_period.items())
+                        if value >= trigger_on_minimum_probability
+                    ),
+                    0.0,
+                )
+
+                # determine the alert class
+                alert_class = classify_alert(
+                    triggered,
+                    likelihood_per_return_period,
+                    classify_alert_on,
+                    alert_on_return_period,
+                    alert_on_minimum_probability,
+                )
+
+                forecast_data_unit = ForecastDataUnit(
+                    adm_level=adm_level,
+                    pcode=pcode,
+                    lead_time=lead_time,
+                    forecasts=forecasts,
+                    triggered=triggered,
+                    return_period=return_period,
+                    alert_class=alert_class,
+                )
+                self.data.forecast_admin.upsert_data_unit(forecast_data_unit)
+
+    def __compute_flood_extent(self):
+        """Compute flood extent raster"""
+        # get country-wide flood extent rasters
+        country = self.data.forecast_admin.country
+
+
+
+        flood_rasters = {}
+        output_dir=self.input_data_path+'/hydrology'
+
+        # Loop through return periods and copy the file 
+        # Here we are going to use a floodextent map developed for the actual event but keepinng the retun period logic 
+        # please replace this as it might not be needed 
+        for rp in  [5,10, 20, 50, 75, 100, 200, 500]:
+            output_filename = f"flood_map_{country.upper()}_RP{rp}.tif"
+            output_path = os.path.join(output_dir, output_filename)
+            shutil.copyfile(self.flood_extent_raster, output_path)
+ 
+        ''' 
+        if os.path.exists(self.flood_extent_raster):
```

Comment:
You can safely remove commented code. It's possible to use git history to find old code.